internal/services/oracle/oracle_autonomous_database_clone_from_database_resource.go

```diff
@@ -0,0 +1,491 @@
+					state.DataStorageSizeInTb = pointer.From(props.DataStorageSizeInTbs)
+					state.DatabaseVersion = pointer.From(props.DbVersion)
+					state.DatabaseVersion = pointer.From(props.DbVersion)
+					state.DatabaseWorkload = pointer.FromEnum(props.DbWorkload)
+					state.DisplayName = pointer.From(props.DisplayName)
+					state.AutoScalingEnabled = pointer.From(props.IsAutoScalingEnabled)
+					state.AutoScalingForStorageEnabled = pointer.From(props.IsAutoScalingForStorageEnabled)
+					state.MtlsConnectionRequired = pointer.From(props.IsMtlsConnectionRequired)
+					state.LicenseModel = pointer.FromEnum(props.LicenseModel)
+					state.NationalCharacterSet = pointer.From(props.NcharacterSet)
+					state.AllowedIpAddresses = pointer.From(props.WhitelistedIPs)
+					state.CustomerContacts = flattenAdbsCustomerContacts(props.CustomerContacts)
+					state.SubnetId = pointer.From(props.SubnetId)
+					state.VnetId = pointer.From(props.VnetId)
+				}
+			}
+
+			return metadata.Encode(&state)
+		},
+	}
+}
+
+func (r AutonomousDatabaseCloneFromDatabaseResource) Delete() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Oracle.OracleClient.AutonomousDatabases
+
+			id, err := autonomousdatabases.ParseAutonomousDatabaseID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			if err := client.DeleteThenPoll(ctx, *id); err != nil {
+				return fmt.Errorf("deleting %s: %+v", *id, err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func (r AutonomousDatabaseCloneFromDatabaseResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return autonomousdatabases.ValidateAutonomousDatabaseID
+}
+
+func (AutonomousDatabaseCloneFromDatabaseResource) CustomizeDiff() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Second,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			if metadata.ResourceDiff == nil {
+				return nil
+			}
+
+			sourceId := metadata.ResourceDiff.Get("source_autonomous_database_id").(string)
+			dbWorkload := metadata.ResourceDiff.Get("database_workload").(string)
+
+			if sourceId == "" || dbWorkload == "" {
+				return nil
+			}
+
+			if metadata.ResourceDiff.Id() != "" {
+				return nil
+			}
+
+			sourceWorkload, err := getSourceWorkloadforClone(ctx, sourceId, metadata)
+			if err != nil {
+				return err
+			}
+
+			targets, exists := workloadMatrixForClone[sourceWorkload]
+			if !exists {
+				return fmt.Errorf("unsupported source workload: %s", sourceWorkload)
+			}
+
+			for _, target := range targets {
+				if dbWorkload == target {
+					return nil
+				}
+			}
+
+			return fmt.Errorf("invalid workload: %s->%s not allowed", sourceWorkload, dbWorkload)
+		},
+	}
+}
+
+var workloadMatrixForClone = map[string][]string{
+	"DW":   {"OLTP", "DW"},
+	"OLTP": {"DW", "OLTP"},
+	"AJD":  {"OLTP", "DW", "APEX"},
+	"APEX": {"AJD", "OLTP", "DW"},
+}
```

Comment:
Could this be moved inside the CustomizeDiff function? I would define it outside if I anticipated needing to reference it from various scopes.